gateway/ln-gateway/src/events.rs

```diff
@@ -22,4 +32,288 @@ pub const ALL_GATEWAY_EVENTS: [EventKind; 11] = [
     DepositConfirmed::KIND,
 ];
 
-// TODO: Add Gateway specific events
+/// Searches through the event log for all events that occurred within the
+/// specified time bounds. Because it is inefficient to search the log
+/// backwards, instead this function traverses the log forwards, but in batches.
+/// All events are appended to an array until the cutoff event where the
+/// timestamp is greater than the start timestamp or the end of the log is hit.
+pub async fn get_events_for_duration(
```

Comment:
I don't know how much should I stress about corectness here. Are people going to be using it for some accounting stuff? Maybe?

Reply:
I think it will mostly be used for an "at a glance" look at what happened in the gateway over some period of time. If there are obvious correctness issues we should fix them, but IMO this doesn't need to be down to the microsecond accurate